bundles/core/src/test/java/com/adobe/cq/commerce/core/components/internal/servlets/CategoryPageRedirectServletTest.java

```diff
@@ -0,0 +1,197 @@
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CategoryPageRedirectServletTest {
+    private CategoryPageRedirectServlet servlet;
+
+    @Mock
+    private SlingHttpServletRequest request;
+
+    @Mock
+    private SlingHttpServletResponse response;
+
+    @Mock
+    private Resource resource;
+
+    @Mock
+    private ResourceResolver resourceResolver;
+
+    @Mock
+    private RequestPathInfo requestPathInfo;
+    private UrlProviderImpl urlProvider;
+    private MockUrlProviderConfiguration config;
+
+    @Mock
+    private PageManagerFactory pageManagerFactory;
+
+    @Mock
+    private PageManager pageManager;
+
+    @Rule
+    public final AemContext context = createContext("/context/jcr-content-redirect-servlet.json");
+
+    private static AemContext createContext(String contentPath) {
+        return new AemContext(
+            (AemContextCallback) context -> {
+                // Load page structure
+                context.load().json(contentPath, "/content");
+            },
+            ResourceResolverType.JCR_MOCK);
+    }
+
+    @Before
+    public void setUp() {
+        servlet = new CategoryPageRedirectServlet();
+
+        when(request.getResourceResolver()).thenReturn(resourceResolver);
+        when(request.getRequestPathInfo()).thenReturn(requestPathInfo);
+        when(request.getResource()).thenReturn(resource);
+
+        Page currentPage = context.pageManager().getPage("/content/venia/us/en");
+        when(pageManager.getContainingPage(resource)).thenReturn(currentPage);
+        when(pageManagerFactory.getPageManager(resourceResolver)).thenReturn(pageManager);
+
+        config = new MockUrlProviderConfiguration();
+        urlProvider = new UrlProviderImpl();
+
+        Whitebox.setInternalState(servlet, "urlProvider", urlProvider);
+        Whitebox.setInternalState(servlet, "pageManagerFactory", pageManagerFactory);
+    }
+
+    @Test
+    public void testMissingProductSuffix() throws IOException {
+        when(requestPathInfo.getSuffix()).thenReturn(null);
+
+        servlet.doGet(request, response);
+        verify(response).sendError(HttpServletResponse.SC_BAD_REQUEST, "Missing redirect suffix.");
+    }
+
+    @Test
+    public void testWrongSuffixLength() throws IOException {
+        when(requestPathInfo.getSuffix()).thenReturn("/some/wrong/suffix");
+
+        servlet.doGet(request, response);
+        verify(response).sendError(HttpServletResponse.SC_BAD_REQUEST, "Redirect suffix has wrong format.");
+    }
+
+    @Test
+    public void testUidMatchingUrlProviderConfig() throws IOException {
+        when(requestPathInfo.getSelectors()).thenReturn(new String[] { CategoryPageRedirectServlet.SELECTOR });
+        when(requestPathInfo.getSuffix()).thenReturn("/test_uid");
+
+        config.setCategoryUrlTemplate("{{page}}.{{uid}}.html");
```

Comment:
While this might work we should not use selectors, lets stick with the patterns we will use: https://github.com/adobe/aem-core-cif-components/tree/issues/CIF-2062/bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/services/urlformats